src/pkg/packager/layout/assemble.go

```diff
@@ -154,11 +155,9 @@ func AssemblePackage(ctx context.Context, pkg v1alpha1.ZarfPackage, packagePath
 		}
 	}
 
-	l.Debug("copying values files to package", "files", pkg.Values.Files)
-	for _, file := range pkg.Values.Files {
-		if err = copyValuesFile(ctx, file, packagePath, buildPath); err != nil {
-			return nil, err
-		}
+	l.Debug("merging values files to package", "files", pkg.Values.Files)
+	if err = mergeAndWriteValuesFile(ctx, pkg.Values.Files, packagePath, buildPath); err != nil {
```

Comment:
Mostly an implementation detail, but I'm 50/50 on whether or not we should include the values.yaml file in the package always or if we should skip writing the file when the package does not set values in the definition. Curious to hear your reasoning behind adding it to the package.

Reply:
I debated this too but decided to just add it to the package so it could be more "expected" to be a part of the package definition in the future (even if for backwards compat reasons that may never happen).  It also simplifies the writing a little since you just iterate through the array (even if it is empty) and write the map without needing additional checks.  The backward compat check on deploy would be needed regardless because of existing packages.

Reply:
Yeah, I think an argument against adding the values.yaml in this situation is that it isn't in line with the current paradigm where the SBOM.tar / zarf.yaml.sig files are not included when those features are not used in the package. However, those files do not have a clear 0 value, whereas the values.yaml does and I think that makes it make sense for me.